src/components/Button/index.tsx

```diff
@@ -1,19 +1,62 @@
 /** @jsx jsx */
 import * as React from "react";
-import { jsx, css } from "@emotion/core";
-import t from "lib/theme";
+import styled from "lib/styled";
+import Box, { BoxProps } from "components/Box";
+import { textSet, TextSetProps } from "onno-react";
 
-interface Props {
-  text?: string;
-}
+type ButtonProps = BoxProps & TextSetProps;
 
-const styles = css({
-  background: t.colors.mojogreen,
-  color: "white",
-});
+interface Props extends ButtonProps {}
 
-const Button: React.SFC<Props> = ({ text, children }) => (
-  <button css={styles}>{text || children}</button>
+const Button: React.FC<Props> = styled(Box)(
+  {
+    backgroundColor: "white",
+    display: "block",
+    border: "none",
+    cursor: "pointer",
+    mixBlendMode: "normal",
+    overflow: "hidden",
+    position: "relative",
+    zIndex: 1,
+    "&:hover": {
+      "&:before": {
+        transform: "translateX(0)",
+      },
+    },
+    "&:before": {
+      backgroundColor: "white",
+      content: "''",
+      height: "100%",
+      left: 0,
+      mixBlendMode: "difference",
+      transform: "translateX(-100%)",
+      position: "absolute",
+      top: 0,
+      transition: "transform .3s ease",
+      width: "100%",
+      willChange: "transform",
+      zIndex: 3,
+    },
+    "&:after": {
+      backgroundColor: "dark",
+      content: "''",
+      left: 0,
+      mixBlendMode: "screen",
+      position: "absolute",
+      height: "100%",
+      top: 0,
+      width: "100%",
+      zIndex: 100,
+    },
+  },
+  textSet,
 );
 
+Button.defaultProps = {
+  as: "button",
```

Comment:
Why not leverage the polymorph helper for this component?

Reply:
This is done because the `Box` already uses the polymorph function, and it is much simpler just to set the `as` prop since the `Button` extends the `Box`.